src/blockdata/script.rs

```diff
@@ -203,6 +198,14 @@ impl std::error::Error for Error {
     }
 }
 
+// Infallible does not implement `Hash` until Rust 1.44
+#[cfg(not(feature = "bitcoinconsensus"))]
+impl core::hash::Hash for BitcoinConsensusError {
+    fn hash<H: core::hash::Hasher>(&self, _: &mut H) {
+        unreachable!()
```

Comment:
This should be `match self._infallible {}` which statically proves it's actually infallible. (Maybe underscore should be removed, IDK if you get warnings bout that or not being created.)